### PR TITLE
Run faas server functions with krun inside the faas server container.

### PR DESCRIPTION
**Pull request by paulpopelka**

### CODE REVIEW

File: kontain-faas/server/Dockerfile [hunk 1]
```diff
@@ -10,23 +10,32 @@
 #
 # Create image with kontain Faas server
 
-FROM alpine:3.7
+#FROM alpine:3.7
+FROM fedora:31
```
[the remark this answers is not in the document]
Comment (paulpopelka) [reply kontain-faas/server/Dockerfile:14]: krun can't be statically linked and I briefly tried to find the needed alpine library but moved on to using fedora.
If alpine is required I can try again.

File: kontain-faas/server/kontain-api-handler.go [hunk 2]
```diff
@@ -41,10 +43,12 @@ func GetCallFunction(url string) (string, error) {
...
 func ApiHandlerExecCallFunction(faasName string) error {
+	containerBaseDir := "run_faas_here"
 	fp := execFileName(faasName)
-	rq := requestFileName(faasName)
-	rp := responseFileName(faasName)
-	execCmd := exec.Command(fp, rq, rp)
+//	rq := requestFileName(faasName)
+//	rp := responseFileName(faasName)
+	containerID := faasName + fmt.Sprintf("_%d", 12)
+	execCmd := exec.Command("/bin/sh", "/opt/kontain/bin/faaskrun.sh", containerBaseDir, fp, pathName, faasName + ".request", faasName + ".response", containerID)
```
Comment (sv641) [inline kontain-faas/server/kontain-api-handler.go:51]: can we use already written function for request and response file names.

File: kontain-faas/server/kontain-api-handler.go [hunk 3]
```diff
@@ -18,10 +18,12 @@ func requestFileName(faasName string) string {
 	return pathName + "/" + faasName + ".request"
 }
 func responseFileName(faasName string) string {
-	return pathName + "/" + faasName + ".response"
+	return pathName + faasName + ".response"
```
Comment (sv641) [inline kontain-faas/server/kontain-api-handler.go:21]: keep these functions uniform(less confusion) / is removed only from some functions.

**Review by sv641 — APPROVED** [review]

File: kontain-faas/server/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (msterin) [reply kontain-faas/server/Dockerfile:14]: what exactly was missing ?

File: kontain-faas/server/kontain-api-handler.go (hunk 2 above)
Comment (msterin) [reply kontain-faas/server/kontain-api-handler.go:51]: And FWIW, constants for bin location ? Not a biggie but does not feel right ...

File: kontain-faas/server/Dockerfile [hunk 4]
```diff
@@ -10,23 +10,32 @@
 #
 # Create image with kontain Faas server
 
-FROM alpine:3.7
+#FROM alpine:3.7
```
Comment (gnode1) [inline kontain-faas/server/Dockerfile:13]: We are using 3.10.0 this days - why are we so much behind here? I understand we switching to Fedora, just curious

Comment (paulpopelka) [reply kontain-faas/server/Dockerfile:13]: I don't know, that is what was there.
I will try switching to 3.10.0 to see what happens.

Comment (paulpopelka) [reply kontain-faas/server/Dockerfile:13]: Tried with alpine 3.10.0 and still see the unresolved symbols.

File: kontain-faas/server/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulpopelka) [reply kontain-faas/server/Dockerfile:14]: Apparently the alpine libraries are missing some symbols that krun references:
```
[paulp@work server]$ kubectl logs --namespace=default kontain-faas-server-d68b7b6bb-2c45p
Error relocating /opt/kontain/bin/krun: copy_file_range: symbol not found
Error relocating /opt/kontain/bin/krun: argp_parse: symbol not found
Error relocating /opt/kontain/bin/krun: statx: symbol not found
Error relocating /opt/kontain/bin/krun: error: symbol not found
Error relocating /opt/kontain/bin/krun: fgetpwent_r: symbol not found
Error relocating /opt/kontain/bin/krun: argp_program_version_hook: symbol not found
krun returned 127
```

File: kontain-faas/server/kontain-api-handler.go (hunk 2 above)
Comment (paulpopelka) [reply kontain-faas/server/kontain-api-handler.go:51]: We don't pass the full path of the request and response files to krun.
We pass the directory containing the files outside of the krun container and then pass the basename of the files to the faaskrun.sh script.
Then the config.json will add the directory into the krun container's mount namespace and km is supplied the container relative full paths for the payload to access.

**Review by johnmuth81 — APPROVED** [review]

**Pull request merged**